intel-hex.js

```diff
@@ -817,6 +817,40 @@ class MemoryMap {
         return memMap;
     }
 
+
+    /**
+     * Returns a new instance of <tt>MemoryMap</tt>, containing only data between
+     * <tt>offset</tt> and <tt>offset</tt> + <tt>length</tt>.
+     *
+     * <br/>
+     * The returned <tt>MemoryMap</tt> might be empty.
+     * <br/>
+     * Internally, this uses <tt>subarray</tt>, so new memory is not allocated.
+     *
+     * @param {Number} offset The start offset
+     * @param {Number} length The lenght of memory map to slice out
+     * @return {MemoryMap}
+     */
+    slice(offset, length){
+        const sliced = new MemoryMap();
+
+        for (let [blockAddr, block] of this) {
+            const blockLength = block.length;
+
+            if ((blockAddr + length) >= offset && blockAddr < (offset + length)) {
+                const sliceStart = Math.max(offset, blockAddr);
+                const sliceEnd = Math.min(offset + length, blockAddr + blockLength);
+                const sliceLength = sliceEnd - sliceStart;
+                const relativeSliceStart = sliceStart - blockAddr;
+
+                if (sliceLength) {
```

Comment:
I don't think it's possible for this to be a negative value, I'll double-check.

Reply:
You're probably right that it can't happen, though following the principle of [don't make me think](https://en.wikipedia.org/wiki/Don%27t_Make_Me_Think) IMO it's best to make the intention explicit.

Reply:
My brain tends to think that lengths cannot ever be negative, but I guess you have a point here.